src/gui_common/ModalManager.cs

```diff
@@ -95,16 +95,23 @@ public bool HideTopMostPopup()
 
         var popup = modalStack.First();
 
-        if (popup.Exclusive && !popup.ExclusiveAllowCloseOnEscape)
-            return false;
+        return HideModal(popup);
+    }
 
-        // This is emitted before closing to allow window using components to differentiate between "cancel" and
-        // "any other reason for closing" in case some logic can be simplified by handling just those two situations.
-        if (popup is CustomWindow dialog)
-            dialog.EmitSignal(nameof(CustomWindow.Cancelled));
+    /// <summary>
+    ///   Attempt to clear all open modals.
+    /// </summary>
+    /// <returns>Returns true if all modals were closed, false otherwise.</returns>
+    public bool ClearModals()
```

Comment:
Would `CloseAllOpenModals` be a better name?

Reply:
I think the current one is already good. This indeed clears `modalStack` by closing all.